Gmsh 2.11 breaks GmshGrids

With Gmsh 2.11 (and maybe earlier), the `GmshGrid` tests fail:

``` python
======================================================================
FAIL: _test (fipy.meshes.gmshMesh.GmshGrid2D)
Doctest: fipy.meshes.gmshMesh.GmshGrid2D._test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/guyer/anaconda/envs/test-environment5/lib/python2.7/doctest.py", line 2226, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for fipy.meshes.gmshMesh.GmshGrid2D._test
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2188, in _test

----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2199, in fipy.meshes.gmshMesh.GmshGrid2D._test
Failed example:
    numerix.allclose(yogmsh._faceAreas, yogrid._faceAreas) # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2202, in fipy.meshes.gmshMesh.GmshGrid2D._test
Failed example:
    yogmsh.cellCenters.value.size == yogrid.cellCenters.value.size # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2207, in fipy.meshes.gmshMesh.GmshGrid2D._test
Failed example:
    mesh.numberOfCells == 4 # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2210, in fipy.meshes.gmshMesh.GmshGrid2D._test
Failed example:
    len(mesh.faceCenters[0]) == 12 # doctest: +GMSH
Expected:
    True
Got:
    False


======================================================================
FAIL: _test (fipy.meshes.gmshMesh.GmshGrid3D)
Doctest: fipy.meshes.gmshMesh.GmshGrid3D._test
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/guyer/anaconda/envs/test-environment5/lib/python2.7/doctest.py", line 2226, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for fipy.meshes.gmshMesh.GmshGrid3D._test
  File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2262, in _test

----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2276, in fipy.meshes.gmshMesh.GmshGrid3D._test
Failed example:
    yogmsh.cellCenters.value.size == yogrid.cellCenters.value.size # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2279, in fipy.meshes.gmshMesh.GmshGrid3D._test
Failed example:
    numerix.allclose(yogmsh._faceAreas, yogrid._faceAreas) # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2282, in fipy.meshes.gmshMesh.GmshGrid3D._test
Failed example:
    numerix.allclose(yogmsh._faceAreas, yogrid._faceAreas) # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2291, in fipy.meshes.gmshMesh.GmshGrid3D._test
Failed example:
    len(mesh.cellCenters.value[0]) == 8 # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2298, in fipy.meshes.gmshMesh.GmshGrid3D._test
Failed example:
    nx.allclose(mesh._faceAreas, faceAreas) # doctest: +GMSH
Expected:
    True
Got:
    False
----------------------------------------------------------------------
File "/Users/guyer/Documents/research/FiPy/fipy/fipy/meshes/gmshMesh.py", line 2308, in fipy.meshes.gmshMesh.GmshGrid3D._test
Failed example:
    nx.allclose(mesh._cellAreas, cellAreas) # doctest: +GMSH
Expected:
    True
Got:
    False


----------------------------------------------------------------------
```

--- a/examples/levelSet/electroChem/gapFillMesh.py
+++ b/examples/levelSet/electroChem/gapFillMesh.py
@@ -47,7 +47,10 @@ only used for the diffusion in the boundary layer.
 
 __docformat__ = 'restructuredtext'
 
+from distutils.version import StrictVersion
+
 from fipy.meshes import Gmsh2D
+from fipy.meshes.gmshMesh import _gmshVersion
 from fipy.meshes import Grid2D
 from fipy.tools import serialComm
 from fipy.tools import parallelComm
@@ -135,7 +138,11 @@ class GapFillMesh(Gmsh2D):
         # Build the fine region mesh.
         self.fineMesh = Grid2D(nx=nx, ny=ny, dx=cellSize, dy=cellSize, communicator=serialComm)
 
-        eps = cellSize / nx / 10
+        if _gmshVersion() < StrictVersion("2.7"):
+            # kludge: must offset cellSize by `eps` to work properly
+            eps = float(cellSize)/(nx * 10)
+        else:
+            eps = 0.
 
         super(GapFillMesh, self).__init__("""
         ny       = %(ny)g;

--- a/fipy/meshes/gmshMesh.py
+++ b/fipy/meshes/gmshMesh.py
@@ -2168,8 +2168,11 @@ class GmshGrid2D(Gmsh2D):
         width  = nx * dx
         numLayers = int(ny / float(dy))
 
-        # kludge: must offset cellSize by `eps` to work properly
-        eps = float(dx)/(nx * 10)
+        if _gmshVersion() < StrictVersion("2.7"):
+            # kludge: must offset cellSize by `eps` to work properly
+            eps = float(dx)/(nx * 10)
+        else:
+            eps = 0.
 
         return """
             ny       = %(ny)g;
@@ -2238,7 +2241,11 @@ class GmshGrid3D(Gmsh3D):
         width  = nx * dx
         depth  = nz * dz
 
-        eps = float(dx)/(nx * 10)
+        if _gmshVersion() < StrictVersion("2.7"):
+            # kludge: must offset cellSize by `eps` to work properly
+            eps = float(dx)/(nx * 10)
+        else:
+            eps = 0.
 
         return """
             ny       = %(ny)g;

